### PR TITLE
Adds OpenTelemetry to the OCR wrapper classes in use

### PR DESCRIPTION
**Pull request by Paethon**

### CODE REVIEW

**Comment by Paethon** [comment]
@CodiumAI-Agent /review
--pr_reviewer.inline_code_comments=true
--pr_reviewer.require_score_review=true
--pr_reviewer.num_code_suggestions="2"

File: ocr_wrapper/bbox_utils.py [hunk 1]
```diff
@@ -169,6 +172,7 @@ def group_overlapping_bboxes(bboxes: list[BBox], threshold: float) -> list[list[
     return groups
 
 
+@tracer.start_as_current_span("merge_bbox_lists")
```
Comment (phschoepf) [inline ocr_wrapper/bbox_utils.py:175]: tracing not needed for the merge_bbox_lists_with_confidences method below?

File: ocr_wrapper/azure.py [hunk 2]
```diff
@@ -109,31 +118,38 @@ def _get_ocr_response(self, img: Image.Image):
...
+                            raise
+                    else:
+                        break
 
             while True:
-                read_result = self.client.get_read_result(operation_id)
-                if read_result.status not in ["notStarted", "running"]:
-                    break
-                time.sleep(0.1)
+                with tracer.start_as_current_span(name="AzureOCR._get_ocr_response.get_read_result") as span:
```
Comment (phschoepf) [inline ocr_wrapper/azure.py:148]: Could cause some log spam. Maybe only record the time taken instead?

**Review by phschoepf — CHANGES_REQUESTED** [review]
Looks good in general. A few open questions, can be merged after they're resolved.

File: ocr_wrapper/azure.py [hunk 3]
```diff
@@ -109,31 +118,38 @@ def _get_ocr_response(self, img: Image.Image):
...
+                    try:
+                        img_stream = BytesIO(img_bytes)
+                        read_response = self.client.read_in_stream(img_stream, raw=True)
+                        read_operation_location = read_response.headers["Operation-Location"]
+                        operation_id = read_operation_location.split("/")[-1]
+                    except (
+                        ComputerVisionOcrErrorException,
+                        ClientRequestError,
+                    ) as e:  # Usually raised because of too many requests or timeout
+                        span.record_exception(e, escaped=True)
```
[the remark this answers is not in the document]
Comment (Paethon) [reply ocr_wrapper/azure.py:131]: Yeah, you are absolutely right! Seems I misread the documentation.

File: ocr_wrapper/bbox_utils.py (hunk 1 above)
Comment (Paethon) [reply ocr_wrapper/bbox_utils.py:175]: No idea what is needed :joy:
But if we have one, doing the other one def. makes sense :grin:

**Pull request merged**